### PR TITLE
test(date): work around missing shadow piercing element in e2e test

### PR DESCRIPTION
**Pull request by paulcpederson**
## Summary

Tests _sometimes_ fail with missing element when you query for an element inside a shadow dom boundary. This only happens on travis. All other environments work fine. For now, I'm just tabbing to the input and executing keyboard commands on the page instead.

### CODE REVIEW

**Comment by paulcpederson** [comment]
@jcfranco ok, this was very involved but I think I got it passing in travis now.

**Review by jcfranco — APPROVED** [review]
Awesome!

**Review by driskull — APPROVED** [review]
👍  nice

**Comment by paulcpederson** [comment]
Just to leave a paper trail for any other devs that encounter travis failing on `>>>` selection, this was logged on stencil e2e a while back https://github.com/ionic-team/stencil/issues/1530, it is actually due to missing functionality in puppeteer: https://github.com/puppeteer/puppeteer/issues/858 , this suggests you use something like:

```js
const input = (
  await page.waitForFunction(() =>
    document.querySelector("calcite-date").shadowRoot.querySelector("calcite-input input")
  )
).asElement();
await input.focus();
```

Basically get an element handle by evaluating a function and returning an actual querySelector on the real shadow root. We should probably make this a test util if it comes up again.

File: src/components/calcite-date/calcite-date.e2e.ts [hunk 1]
```diff
@@ -7,25 +7,27 @@ describe("calcite-date", () => {
   it("fires a calciteDateChange event on change", async () => {
     const page = await newE2EPage();
     await page.setContent("<calcite-date></calcite-date>");
-    const input = await page.find("calcite-date >>> calcite-input");
+    await page.keyboard.press("Tab");
     const changedEvent = await page.spyOnEvent("calciteDateChange");
-    await input.callMethod("setFocus");
     // have to wait for transition
     await new Promise((res) => setTimeout(() => res(true), 200));
```
[the remark this answers is not in the document]
Comment (paulcpederson) [reply src/components/calcite-date/calcite-date.e2e.ts:18]: isn't waitFor deprecated? Or have I imagined seeing that 10,000 times in the logs? 😆

[the remark this answers is not in the document]
Comment (jcfranco) [reply src/components/calcite-date/calcite-date.e2e.ts:18]: Yes, but the types haven't been updated for us to use `waitForTimeout` (https://github.com/Esri/calcite-components/pull/1044).

**Pull request merged**